modules/dop/services/cdp/cdp.go

```diff
@@ -276,3 +279,16 @@ func (cdp *CDP) getFailedTaskLogs(pipelineDetail *apistructs.PipelineDetailDTO)
 	}
 	return failedDetailLogs, nil
 }
+
+// truncateLog return logs with given lines
+func truncateLog(sourceLogs string, lines int) string {
+	if lines <= 0 {
+		return sourceLogs
+	}
+	logSplits := strings.Split(sourceLogs, "\n")
+	if len(logSplits) <= lines {
+		return sourceLogs
+	}
+	truncateLogs := logSplits[:lines]
+	return strings.Join(truncateLogs, "\n")
+}
```

Comment:
do we also need truncate columns?

Reply:
added, set default max columns 1024, 1024 byte X 10 lines, so the max log size is limitted 10k